File: auctionsite-ui/src/models/UserProfileModel.ts
import { Gender } from "./enum/Gender";

export default interface UserProfileModel {
    id: string,
    userName: string,
    firstName?: string,
    lastName?: string,
    phoneNumber?: string,
    city?: string,
    postalCode?: string,
    country?: string,
    address?: string,
    gender?: Gender
}

File: AuctionSite.Models/User/UserProfileModel.cs
using AuctionSite.Domain.Enum;

namespace AuctionSite.Models.User
{
    public class UserProfileModel
    {
        public string Id { get; set; }
        public string UserName { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? PhoneNumber { get; set; }
        public string? City { get; set; }
        public string? PostalCode { get; set; }
        public string? Country { get; set; }
        public string? Address { get; set; }
        public Gender? Gender { get; set; }
    }
}
